packages/tools/src/lib/build-cache/github/GitHubBuildCache.ts

```diff
@@ -1,116 +1,92 @@
-import { color } from '../../color.js';
-import { getGitRemote } from '../../git.js';
-import logger from '../../logger.js';
-import type { spinner } from '../../prompts.js';
-import type {
-  LocalArtifact,
-  RemoteArtifact,
-  RemoteBuildCache,
-} from '../common.js';
-import { getLocalArtifactPath } from '../common.js';
+import { RnefError } from '../../error.js';
+import type { RemoteArtifact, RemoteBuildCache } from '../common.js';
 import {
-  downloadGitHubArtifact,
+  deleteGitHubArtifacts,
   fetchGitHubArtifactsByName,
 } from './artifacts.js';
 import type { GitHubRepoDetails } from './config.js';
-import {
-  detectGitHubRepoDetails,
-  getGitHubToken,
-  promptForGitHubToken,
-} from './config.js';
+import { detectGitHubRepoDetails } from './config.js';
 
 export class GitHubBuildCache implements RemoteBuildCache {
   name = 'GitHub';
   repoDetails: GitHubRepoDetails | null = null;
 
-  async detectRepoDetails() {
-    const gitRemote = await getGitRemote();
-    this.repoDetails = gitRemote
-      ? await detectGitHubRepoDetails(gitRemote)
-      : null;
-    if (!this.repoDetails) {
-      return null;
+  constructor(config?: { owner: string; repository: string; token: string }) {
+    if (config) {
+      this.repoDetails = {
+        owner: config.owner,
+        repository: config.repository,
+        token: config.token,
+      };
     }
-    if (!getGitHubToken()) {
-      logger.warn(
-        `No GitHub Personal Access Token found necessary to download cached builds.
-Please generate one at: ${color.cyan('https://github.com/settings/tokens')}
-Include "repo", "workflow", and "read:org" permissions.`
-      );
-      await promptForGitHubToken();
+  }
+
+  async getRepoDetails() {
+    if (!this.repoDetails) {
+      this.repoDetails = await detectGitHubRepoDetails();
     }
     return this.repoDetails;
   }
 
-  async query({
+  async list({
     artifactName,
+    limit,
   }: {
-    artifactName: string;
-  }): Promise<RemoteArtifact | null> {
-    const repoDetails = await this.detectRepoDetails();
-    if (!getGitHubToken()) {
-      logger.warn(`No GitHub Personal Access Token found.`);
-      return null;
-    }
-
-    if (!repoDetails) {
-      return null;
-    }
-
+    artifactName?: string;
+    limit?: number;
+  }): Promise<RemoteArtifact[]> {
+    const repoDetails = await this.getRepoDetails();
     const artifacts = await fetchGitHubArtifactsByName(
       artifactName,
-      repoDetails
+      repoDetails,
+      limit
     );
+    return artifacts.map((artifact) => ({
+      name: artifact.name,
+      url: artifact.downloadUrl,
+    }));
+  }
+
+  async download({
+    artifactName,
+  }: {
+    artifactName: string;
+  }): Promise<Response> {
+    const repoDetails = await this.getRepoDetails();
+    const artifacts = await this.list({ artifactName });
     if (artifacts.length === 0) {
-      return null;
+      throw new RnefError(`No artifact found with name "${artifactName}"`);
```

Comment:
so someone building their own cache provider should also use `RnefError`? we should document this or mention in the JSDoc

Reply:
For better error logging it would be preferable to use RnefError, but one can use Error

Reply:
Let's please mention it somewhere. Is `RnefError` exposed in public API?